homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -84,6 +84,57 @@
 ATTR_FEATURES = 'features'
 ATTR_VOLUME = 'volume'
 
+# Map attributes to properties of the state object
+AVAILABLE_ATTRIBUTES_AIRPURIFIER_COMMON = {
+    ATTR_TEMPERATURE: 'temperature',
+    ATTR_HUMIDITY: 'humidity',
+    ATTR_AIR_QUALITY_INDEX: 'aqi',
+    ATTR_MODE: 'mode',
+    ATTR_FILTER_HOURS_USED: 'filter_hours_used',
+    ATTR_FILTER_LIFE: 'filter_life_remaining',
+    ATTR_FAVORITE_LEVEL: 'favorite_level',
+    ATTR_CHILD_LOCK: 'child_lock',
+    ATTR_LED: 'led',
+    ATTR_MOTOR_SPEED: 'motor_speed',
+    ATTR_AVERAGE_AIR_QUALITY_INDEX: 'average_aqi',
+    ATTR_PURIFY_VOLUME: 'purify_volume',
+    ATTR_LEARN_MODE: 'learn_mode',
+    ATTR_SLEEP_TIME: 'sleep_time',
+    ATTR_SLEEP_LEARN_COUNT: 'sleep_mode_learn_count',
+    ATTR_EXTRA_FEATURES: 'extra_features',
+    ATTR_TURBO_MODE_SUPPORTED: 'turbo_mode_supported',
+    ATTR_AUTO_DETECT: 'auto_detect',
+}
+
+AVAILABLE_ATTRIBUTES_AIRPURIFIER = \
+    dict(AVAILABLE_ATTRIBUTES_AIRPURIFIER_COMMON)
+AVAILABLE_ATTRIBUTES_AIRPURIFIER.update({
+    ATTR_BUZZER: 'buzzer',
+    ATTR_LED_BRIGHTNESS: 'led_brightness',
+    ATTR_SLEEP_MODE: 'sleep_mode',
+})
```

Comment:
Considering minimum python version being 3.5 now, I think the following would also work to simplify it a bit more:
```
AVAILABLE_ATTRIBUTES_AIRPURIFIER = {**AVAILABLE_ATTRIBUTES_AIRPURIFIER_COMMON,
                                    ATTR_BUZZER: 'buzzer',
}
```